dev/CommandBarFlyout/InteractionTests/CommandBarFlyoutTests.cs

```diff
@@ -992,5 +992,86 @@ public void VerifyDynamicSecondaryCommandLabel()
                 Verify.AreEqual(finalBoundingRectangle.Height, initialBoundingRectangle.Height);
             }
         }
+
+        [TestMethod]
+        public void VerifyIsFlyoutKeyboardAccessibleWithNoPrimaryCommands()
+        {
+            if (PlatformConfiguration.IsOSVersionLessThan(OSVersion.Redstone2))
+            {
+                Log.Warning("Test is disabled pre-RS2 because CommandBarFlyout is not supported pre-RS2");
+                return;
+            }
+
+            using (var setup = new CommandBarFlyoutTestSetupHelper())
+            {
+                Button showCommandBarFlyoutButtonWithNoPrimaryCommands = FindElement.ByName<Button>("Show CommandBarFlyout with no primary commands");
+
+                Log.Comment("Tap on a button to show the CommandBarFlyout.");
+                showCommandBarFlyoutButtonWithNoPrimaryCommands.InvokeAndWait();
+
+                Button undoButton6 = FindElement.ById<Button>("UndoButton6");
+                var undoButtonElement = AutomationElement.FocusedElement;
+                Verify.AreEqual(undoButtonElement.Current.AutomationId, undoButton6.AutomationId);
+                
+                KeyboardHelper.PressKey(Key.Up);
+                Wait.ForIdle();
+                
+                if (PlatformConfiguration.IsOSVersionLessThan(OSVersion.Redstone3))
+                {
+                    // rs2 does not loop through menuflyout items, so focus stays on the first button.
+                    Log.Comment("Press Up key to make sure it does not go to hidden controls.");
+                    Verify.AreEqual(undoButtonElement.Current.AutomationId, undoButton6.AutomationId);
+                }
+                else
+                {
+                    Log.Comment("Press Up key to make sure commands loops focus to last secondary command: Favorite.");
+                    Button favoriteToggleButton6 = FindElement.ById<Button>("FavoriteToggleButton6");
+                    var favoriteToggleElement = AutomationElement.FocusedElement;
+                    Verify.AreEqual(favoriteToggleElement.Current.AutomationId, favoriteToggleButton6.AutomationId);
+                }
+            }
+        }
+
+        [TestMethod]
+        public void VerifyAddRemovePrimaryCommands()
+        {
```

Comment:
This test doesn't test the removal of a primary command, only the addition of one.

More importantly, I wouldn't expect the behavior to fail when you change the number of commands in the CBF while it is closed.  The bug I called out in the product code would only happen if you change the number of commands while the flyout is open.

Reply:
I don't think there is a scenario where the number of commands will change while the flyout is open, since CBF is a flyout and transient by design.

Reply:
Discussed offline. Test case is updated to be adding PrimaryCommands dynamically.

Reply:
For completeness removing the last primary command could also be tested. It uses what looks like the same code path though so maybe it is fine.